### DISCUSSION TITLE
Strange hamburger menu behaviour

### DISCUSSION THREAD

**Discussion by kbl** [mmistakes/minimal-mistakes#2569] [Q&A]
First – thanks for your magnificent work, minimal mistakes is the thing that makes me stay with jekyll :)

I've observed strange behavior of hamburger menu. Depending on some thing that I'm not aware of sometimes JS is executed properly and hamburger menu is hidden, and sometimes I can see in console `TypeError: $(...).fitVids is not a function` stopping JS exeucution/leaving hamburger menu visible.

```
jQuery.Deferred exception: $(...).fitVids is not a function @https://www.samouczekprogramisty.pl/assets/js/main.min.js:9:24516
f@https://www.samouczekprogramisty.pl/assets/js/main.min.js:7:9959
i/</d<@https://www.samouczekprogramisty.pl/assets/js/main.min.js:7:10263
 undefined main.min.js:7:11541
```

Below are two screenshots of the same page. On first one everything works as intended:
![Hamburger menu hidden](https://user-images.githubusercontent.com/73585/84711155-efebff00-af65-11ea-8795-2bd63daee8c9.png)

On second one JS execution fails, which also causes search being unusable:
![Hamburger menu visible](https://user-images.githubusercontent.com/73585/84711106-d5b22100-af65-11ea-865e-4fc741eb1647.png)

Post used here is just an example. I can see similar behavior on other posts as well.

What is causing that strange behavior? What can I do to fix it?

Page is generated locally and pushed to github and hosted from github pages.
Jekyll: 4.0.1
minimal-mistakes-jekyll: 4.19.3
Source code available here: https://github.com/SamouczekProgramisty/samouczekprogramisty.github.io/tree/source
Website available here: https://www.samouczekprogramisty.pl/wzorzec-projektowy-dekorator/
Web browser: Firefox 76.0.1
OS: Ubuntu 18.04

**Comment by mmistakes** [comment]
Not sure. Doesn't look like you customized the theme's JavaScript, though I haven't done a deep dive into your repo. But that would have been the first thing I'd suspect since if you're getting JavaScript errors in the console that could interfere with search and the menu toggle.

I clicked around your site and didn't see any issues. Is there a specific page that you reliably see the FitVids error for? I'm not seeing any console errors.

Since you customized the theme and the issue doesn't appear to be in the MM demo site it would appear to be something unique to your fork. Are you sure all the layouts and includes are current with the theme's files? And that you're not loading any additional scripts that might interfere with `main.min.js`?

**Comment by kbl** [comment]
Thanks, I'll need to look deeper into that – will update the ticket in case I find something.
